lib/ansible/module_utils/facts/system/pkg_mgr.py

```diff
@@ -85,13 +85,17 @@ def _check_rh_versions(self, pkg_mgr_name, collected_facts):
                 break
 
         try:
-            distro_major_ver = int(collected_facts['ansible_distribution_major_version'])
+            major_version = collected_facts['ansible_distribution_major_version']
+            if collected_facts['ansible_distribution'] == 'Kylin Linux Advanced Server':
+                major_version = major_version[1:] if major_version[0] == 'V' else major_version
```

Comment:
```suggestion
            if collected_facts['ansible_distribution'] == 'Kylin Linux Advanced Server':
                major_version = major_version.lstrip('V')
```

Reply:
This is indeed a better way to handle it, thanks.